### ISSUE TITLE
Markdown assets pulled from S3 bump into transformer-remark

### ISSUE THREAD

**Issue by rlenkov** [gatsbyjs/gatsby#25964]
## Summary
I have markdown files that I'm accessing using gatsby-source-s3-asset plugin. These get parsed from the GraphQL schema by  gatsby-transformer-remark plugin before their download, resulting in some invalid markdown nodes with empty attributes.
## Relevant information
I'm using the [blog starter](https://www.gatsbyjs.org/starters/gatsbyjs/gatsby-starter-blog/) with some changes: I have some markdown files and images in an S3 bucket that I'm accessing using gatsby-source-s3-asset plugin. I also have gatsby-transformer-remark plugin with the default setup of gatsby-starter-blog, but I've deleted the local blogposts in the repo (as I'm fetching my md files from S3).
The source-s3 plugin actually warns me in build time like this:
```
warn Warning: gatsby-source-s3-asset does not download node contents.

    This probably happened because some transformer plugin such as
    gatsby-transformer-remark is trying to parse the contents of the asset
    at URL https://*****-****.s3.******-*.amazonaws.com/folder-1/index.md.

    If you want S3 assets to be further processed you must use another plugin
    that actually downloads the files to your local system.
```
> which is not a problem as I take care of the download, see below

After the build I end up having invalid markdownRemark nodes with empty fields. It pretty much looks like these invalid nodes are made by gatsby-transformer-remark trying to parse the s3Asset nodes where the asset has markdown extension.
I actually download the S3 markdown files too in an `onCreateNode` step and they get created as markdownRemark nodes as well with proper attributes, but I still have the invalid nodes too as duplicates.

I tried to add a cheeky `deleteNode` action but the build kicked me in the back, saying `The plugin "default-site-plugin" deleted a node of a type owned by another plugin.`. Any ideas how should I handle this? (filtering the invalid nodes on the createPages call is my last resort, as I don't like that my graphQL schema is messed up)
See my `gatsby-node.js` below.

### Environment (if relevant)
  System:
    OS: Linux 5.3 Ubuntu 18.04.4 LTS (Bionic Beaver)
    CPU: (4) x64 Intel(R) Core(TM) i5-4210U CPU @ 1.70GHz
    Shell: 4.4.20 - /bin/bash
  Binaries:
    Node: 12.18.2 - /usr/bin/node
    Yarn: 1.22.4 - ~/.yarn/bin/yarn
    npm: 6.14.5 - /usr/bin/npm
  Languages:
    Python: 2.7.17 - /usr/bin/python
  Browsers:
    Chrome: 84.0.4147.89
    Firefox: 78.0.2
  npmPackages:
    gatsby: ^2.23.12 => 2.23.12
    gatsby-image: ^2.4.9 => 2.4.9
    gatsby-plugin-feed: ^2.5.7 => 2.5.7
    gatsby-plugin-google-analytics: ^2.3.6 => 2.3.6
    gatsby-plugin-manifest: ^2.4.14 => 2.4.14
    gatsby-plugin-offline: ^3.2.13 => 3.2.13
    gatsby-plugin-prefetch-google-fonts: ^1.4.3 => 1.4.3
    gatsby-plugin-react-helmet: ^3.3.6 => 3.3.6
    gatsby-plugin-sass: ^2.3.12 => 2.3.12
    gatsby-plugin-sharp: ^2.6.14 => 2.6.14
    gatsby-remark-copy-linked-files: ^2.3.7 => 2.3.7
    gatsby-remark-images: ^3.3.14 => 3.3.14
    gatsby-remark-prismjs: ^3.5.6 => 3.5.6
    gatsby-remark-responsive-iframe: ^2.4.7 => 2.4.7
    gatsby-remark-smartypants: ^2.3.6 => 2.3.6
    gatsby-source-filesystem: ^2.3.14 => 2.3.14
    gatsby-source-s3-asset: ^1.4.0 => 1.4.0
    gatsby-transformer-remark: ^2.8.20 => 2.8.20
    gatsby-transformer-sharp: ^2.5.7 => 2.5.7
  npmGlobalPackages:
    gatsby-cli: 2.12.59

### File contents (if changed)
`gatsby-node.js`:
```javascript
const path = require(`path`)
const {
    createRemoteFileNode,
} = require(`gatsby-source-filesystem`)

// a createPage export using markdown data comes here if this is resolved.

exports.onCreateNode = ({ node, getCache, createNodeId, actions }) => {
    const { createNode, deleteNode } = actions

    if (node.internal.type === `MarkdownRemark`) {
        if (!node.rawMarkdownBody) {
            deleteNode({node: node})
        }
    }

    if (node.internal.type === `S3Asset`) {
        let fileNode
        try {
            fileNode = createRemoteFileNode({
                url: node.url,
                parentNodeId: node.id,
                getCache,
                createNode,
                createNodeId,
            })
        } catch (e) {
            console.log('shit happened')
            console.log(e)
        }

        // Adds a field `localFile` to the node
        // ___NODE appendix tells Gatsby that this field will link to another node
        if (fileNode) {
            node.localFile___NODE = fileNode.id
        }
    }
}
```

**Comment by vladar** [comment]
I think the warning of `gatsby-source-s3-asset` plugin sums it up pretty well.

The problem is that when `gatsby-source-s3-asset` creates nodes, it assigns `node.internal.mediaType` to it [here](https://github.com/kilburn/gatsby-source-s3-asset/blob/c5edd230f0fd2b2a8bb10e171208c45a25fe2f37/src/utils.ts#L97). Remark plugin hooks to `onCreateNode` too and checks for this media type.

Then it tries loading contents of the node (using [`loadNodeContent`](https://github.com/kilburn/gatsby-source-s3-asset/blob/master/src/index.ts)  function of the `gatsby-source-s3-asset` plugin). And here you get this error.

This is an intentional choice of `gatsby-source-s3-asset`. So I guess your options are:

1. Contribute an actual `loadNodeContent` function in their repo
2. Suggest [removing this line](https://github.com/kilburn/gatsby-source-s3-asset/blob/c5edd230f0fd2b2a8bb10e171208c45a25fe2f37/src/utils.ts#L97) (as `mediaType` is an internal mechanism for other plugins to detect specific kinds of nodes, it doesn't make much sense for their plugin as they don't download actual content)
3. Or use another plugin as they suggest (maybe fork?).

**Comment by rlenkov** [comment]
Pretty much what I've thought. Thank you very much for your answer!

I've realized that my approach is flawed anyway :) Actually I don't need the aws assets as nodes in my schema. I just need them at the right place in the right time during build.

What I did is that I deleted the `gatsby-source-s3-asset` plugin and now I simply download my assets using `aws-sdk` and I copy them under `/content/blog` in a `onPreBootstrap` step, so that `gatsby-transformer-remark` can pick them up during build. :)
